src/solver/hydro/management/daily.cpp

```diff
@@ -391,7 +391,7 @@ inline void HydroManagement::prepareDailyOptimalGenerations(Solver::Variable::St
             uint firstDay = study.calendar.months[simulationMonth].daysYear.first;
             uint endDay = firstDay + daysPerMonth;
 
-            DONNEES_MENSUELLES problem = H2O_J_Instanciation();
+            DONNEES_MENSUELLES& problem = *H2O_J_Instanciation();
```

Comment:
This is dangerous. De-referencing a pointer to keep it in a reference could lead to a dangling reference. Example
```
  int& p = *(new int(5));
  delete &p;
  std::cout << p << std::endl; // invalid read / segfault
```
When we have a pointer, we keep in mind that it may be `nullptr`. There is no such thing for references, so you have to be careful.